File: STUDY/PYTHON/12_Proyecto_Integrador/ejercicios_basico.py
"""
Ejercicios basico - Proyecto integrador
"""

def add_task(tareas, titulo):
    """Agrega tarea con id incremental."""
    # TODO: implementa la logica
    return tareas


def list_tasks(tareas):
    """Retorna lista de tareas."""
    # TODO: implementa la logica
    return []



def mark_done(tareas, task_id):
    """Marca tarea como hecha y retorna True/False."""
    # TODO: implementa la logica
    return False

def main():
    tareas = []
    tareas = add_task(tareas, "leer")
    print("list:", list_tasks(tareas))


if __name__ == "__main__":
    main()

File: STUDY/RUST/11_Proyectos_Complejos/ejercicios_basico.rs
// ============================================
// EJERCICIOS BASICO - PROYECTOS COMPLEJOS
// ============================================

#[derive(Debug)]
struct Tarea {
    id: u32,
    titulo: String,
}

fn agregar(tareas: &mut Vec<Tarea>, titulo: &str) {
    // TODO: agrega tarea con id incremental
}


fn listar_ids(tareas: &Vec<Tarea>) -> Vec<u32> {
    // TODO: retorna ids
    Vec::new()
}

fn main() {
    let mut tareas = Vec::new();
    agregar(&mut tareas, "leer");
    println!("tareas: {:?}", tareas);
}

